SU2_DOT/src/SU2_DOT.cpp

```diff
@@ -957,3 +967,99 @@ void SetSensitivity_Files(CGeometry ***geometry, CConfig **config, unsigned shor
   }
 
 }
+
+void DerivativeTreatment_MeshSensitivity(CGeometry *geometry, CConfig *config, CVolumetricMovement *grid_movement) {
+
+  int rank = SU2_MPI::GetRank();
+
+  /*--- Warning if choosen mode is unsupported.
+   * This is the default option if the user has not specified a mode in the config file. ---*/
+  if (config->GetSobMode() == ENUM_SOBOLEV_MODUS::NONE) {
+    if (rank == MASTER_NODE)  cout << "Unsupported operation modus for the Sobolev Smoothing Solver." << endl;
+    return;
+  }
+
+  /*-- Construct the smoothing solver and numerics ---*/
+  CSolver* solver = new CGradientSmoothingSolver(geometry, config);
+  CNumerics* numerics;
+  if (config->GetSmoothOnSurface()) {
+    numerics = new CGradSmoothing(geometry->GetnDim()-1, config);
+  } else {
+    numerics = new CGradSmoothing(geometry->GetnDim(), config);
+  }
```

Comment:
This is leaking.
```suggestion
  std::unique_ptr<CSolver> solver = new CGradientSmoothingSolver(geometry, config);
  std::unique_ptr<CNumerics> numerics;
  if (config->GetSmoothOnSurface()) {
    numerics = new CGradSmoothing(geometry->GetnDim()-1, config);
  } else {
    numerics = new CGradSmoothing(geometry->GetnDim(), config);
  }
```

Reply:
I looked into this.
Changing CSolver* => std::unique_ptr<CSolver> is fine.
Changing CNumerics* => std::unique_ptr<CNumerics> leads to issues, since 'numerics' is handed as an input to functions from the CSolver interface and there is no conversion. I don't want to change/ overload existing CSolver functions for other arguments. Therefore, I suggest just adding 
'delete numercs;' 
at the end of the routine?

Reply:
to pass it to other functions you would do `numerics.get()` which gives you a plain pointer

Reply:
Ok thanks for the hint.